lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -710,10 +716,21 @@ export class UserAgentApplication {
                         return null;
                     });
             }
-        }).catch((error: AuthError) => {
-            this.cacheStorage.resetTempCacheItems(request.state);
-            throw error;
-        });
+        })
+            .then(res => {
+                apiEvent.wasSuccessful = true;
+                return res;
+            })
+            .catch((error: AuthError) => {
+                this.cacheStorage.resetTempCacheItems(request.state);
+                apiEvent.apiErrorCode = error.errorCode;
+                apiEvent.wasSuccessful = false;
```

Comment:
make this a function on apiEvent instead

Reply:
same comment on changing design in another pr